sphinx/ext/autodoc.py

```diff
@@ -171,13 +171,15 @@ def __init__(self, names):
             #     set(['a', 'd'])
             self.base_packages.add(n.split('.')[0])
         self.mocked_modules = []  # type: List[str]
-        self.orig_meta_path = sys.meta_path
         # enable hook by adding itself to meta_path
         sys.meta_path = sys.meta_path + [self]
 
     def disable(self):
         # restore original meta_path to disable import hook
-        sys.meta_path = self.orig_meta_path
+        for idx, importer in enumerate(sys.meta_path):
+            if importer is self:
+                del sys.meta_path[idx]
+                break
```

Comment:
Ah, I misunderstood it. Now I understand clearly.
OK, let's use "list comprehensions" as your suggestion.